cf-custom-resources/lib/dns-cert-validator.js

```diff
@@ -161,19 +192,197 @@ const requestCertificate = async function (
   return reqCertResponse.CertificateArn;
 };
 
+const updateHostedZoneRecords = async function (
+  action,
+  options,
+  envName,
+  appName,
+  domainName,
+  envRoute53,
+  appRoute53,
+  envHostedZoneId
+) {
+  const promises = [];
+  for (const option of options) {
+    switch (option.DomainName) {
+      case `${envName}.${appName}.${domainName}`:
+        promises.push(
+          validateDomain({
+            route53: envRoute53,
+            record: option.ResourceRecord,
+            action: action,
+            domainName: "",
+            hostedZoneId: envHostedZoneId,
+          })
+        );
+        break;
+      case `${appName}.${domainName}`:
+        promises.push(
+          validateDomain({
+            route53: appRoute53,
+            record: option.ResourceRecord,
+            action: action,
+            domainName: `${appName}.${domainName}`,
+          })
+        );
+        break;
+      case domainName:
+        promises.push(
+          validateDomain({
+            route53: appRoute53,
+            record: option.ResourceRecord,
+            action: action,
+            domainName: domainName,
+          })
+        );
+        break;
+    }
+  }
+  return Promise.all(promises);
+};
+
+const deleteHostedZoneRecords = async function (
+  options,
+  envName,
+  appName,
+  domainName,
+  envRoute53,
+  appRoute53,
+  acm,
+  envHostedZoneId
+) {
+  let isLegacyCert = false;
+  // Legacy cert only has two DomainValidationOptions.
+  if (options.length <= 2) {
```

Comment:
wut why two, wouldn't it be just 1 with the `env.app.domain` option? Can we expand on the comment here explaining which two.

Reply:
comment added